File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        r,l = 0, len(numbers) -1
        while r < l:
            res = numbers[l] + numbers[r] 
            if res > target:
                l -= 1
            elif res < target:
                r += 1
            else:
                return [r+1,l+1]
        

File: Data Structures & Algorithms/string-encode-and-decode/submission-2.cs
public class Solution {

        public string Encode(IList<string> strs) {
            string res = "";

            foreach ( string s in strs) {
                int num = s.Length;
                res += $"{num}#{s}";
            }

            return res;
        } 

        public List<string> Decode(string s) {
            if (s == ""){
                return new List<string> ();
            }
            List<string> res = new List<string>();

            int i=0;
            int endNum = i;
            do {
                endNum = i;
                while (s[endNum] !='#') {
                    endNum++;
                    
                }
                int nextWordLength = int.Parse(s.Substring(i,endNum - i));
                int end = nextWordLength;
               
                Console.WriteLine(s.Substring(endNum+1, end));
                
                res.Add(s.Substring(endNum+1 , end));
                i += nextWordLength + (endNum - i)+1;
            } while (i < s.Length);

            return res;
        } 
}
